### ISSUE TITLE
run commands as hooks

### ISSUE THREAD

**Issue by eurochriskelly** [fgeorges/mlproj-core#15]
There are certain commands that one would like to run before and/or after other commands. It would be nice if I could define a custom command and have it run automatically as a "post-" or "pre-" command hook.

e.g.
```
hooks: [{
    on: ['deploy', 'watch'],
    before: [ 'clearTDE' ],
    after: [ 'initTDE' ]
}],
commands: {
    clearTDE: () => { /** clear templates **/ },
    loadTDE: () => { /** load templates**/ }
}
```

**Comment by fgeorges** [comment]
Sounds like a good idea in the general case.  In this one though:

- TDE is something we would like to support first-hand (is it possible you share `clearTDE` and `loadTDE` once complete?)
- so it this very case, setting up TDE before and or after deploying modules might be supported automatically if declared as such

:-)

**Comment by eurochriskelly** [comment]
Yeah, I can share those. I made a separate suggestion about plugins before reading this where I also used tde as an example :)
